packages/flame/lib/src/text/styles/document_style.dart

```diff
@@ -0,0 +1,137 @@
+    BlockStyle? header1Style,
+    BlockStyle? header2Style,
+    BlockStyle? header3Style,
+  }) : padding = padding ?? EdgeInsets.zero {
+    backgroundStyle = acquire(background);
+    this.paragraphStyle = acquire(paragraphStyle ?? defaultParagraphStyle)!;
+    this.header1Style = acquire(header1Style ?? defaultHeader1Style)!;
+    this.header2Style = acquire(header2Style ?? defaultHeader2Style)!;
+    this.header3Style = acquire(header3Style ?? defaultHeader3Style)!;
+  }
+
+  /// Outer width of the document page.
+  ///
+  /// This width is the distance between the left edge of the left border, and
+  /// the right edge of the right border. Thus, it corresponds to the
+  /// "border-box" box sizing model in HTML.
+  ///
+  /// If this property is `null`, then the page width must be provided when
+  /// formatting a document.
+  final double? width;
+
+  /// Outer height of the document page.
+  ///
+  /// If the [overflow] property is [Overflow.expand], then the height parameter
+  /// is treated as a minimum height, in other cases the height is obeyed
+  /// strictly.
+  ///
+  /// If this property is `null`, then the page height must be provided when
+  /// formatting a document (except for the overflow-expand mode, when the
+  /// value of [height] defaults to 0).
+  final double? height;
+
+  /// Behavior of the document when the amount of content that needs to be laid
+  /// out exceeds the provided [height]. See the [Overflow] enum description for
+  /// more details.
+  final Overflow overflow = Overflow.expand;
+
+  /// The distance from the outer edges of the page to the inner content box of
+  /// the document.
+  ///
+  /// Note that the padding is computed from the outer edge of the page, unlike
+  /// HTML where it is computed from the inner edge of the border box.
+  ///
+  /// If the document's horizontal padding exceeds its width, an exception will
+  /// be thrown.
+  final EdgeInsets padding;
+
+  /// If present, describes what kind of background and borders to draw for the
+  /// document page(s).
+  late final BackgroundStyle? backgroundStyle;
+
+  /// Style for paragraph nodes in the document.
+  late final BlockStyle paragraphStyle;
+
+  /// Style for level-1 headers.
+  late final BlockStyle header1Style;
+
+  /// Style for level-2 headers.
+  late final BlockStyle header2Style;
+
+  /// Style for level-3 headers.
+  late final BlockStyle header3Style;
+
+  static BlockStyle defaultParagraphStyle = BlockStyle();
+  static BlockStyle defaultHeader1Style = BlockStyle();
+  static BlockStyle defaultHeader2Style = BlockStyle();
+  static BlockStyle defaultHeader3Style = BlockStyle();
+
+  @override
+  DocumentStyle clone() => copyWith();
+
+  DocumentStyle copyWith({
+    double? width,
+    double? height,
+    EdgeInsets? padding,
+    BackgroundStyle? background,
+    BlockStyle? paragraphStyle,
+    BlockStyle? header1Style,
+    BlockStyle? header2Style,
+    BlockStyle? header3Style,
```

Comment:
Could this be a list instead, supporting more than 3 header styles?

Reply:
We could easily add `header4Style` and `header5Style` if you think a more deep documents would be needed. However, I don't want to have _arbitrarily_ deep header levels -- that way lies madness...

Also, bear in mind that these header styles will all have defaults, like `Style(fontScale: 2.0, margins: ...)`. So that by default a header would look like a header, not just a plain text. Of course, these would be overridable if the user don't like the defaults. But they may only have an objection against `header3Style`, say, and be fine with `header1Style` and `header2Style`. Having the styles as separate properties makes overriding them easier.

It also seem to be more in line with how people usually think about the document structure. There is `<H1>`, `<H2>`, and `<H3>` to designate nested sections. But it's not very common to think of them as a "list of headers of consecutive levels".

Reply:
Maybe we should at least have 1-6 then, to be in line with HTML and flutter.

Reply:
Ok, let's do that